### PR TITLE
Dove: remove #![allow(dead_code)] in test

### PR DESCRIPTION
**Pull request by vladimirovmm**

### CODE REVIEW

File: dove/src/lib.rs [hunk 1]
```diff
@@ -17,3 +17,5 @@ pub mod context;
 pub mod index;
 /// Dove configuration.
 pub mod manifest;
+/// for tests
+pub mod tests_helper;
```
Comment (vitvakatu) [inline dove/src/lib.rs:21]: Shouldn't this module be `#[cfg(test)]`?

File: dove/src/tests_helper.rs [hunk 2]
```diff
@@ -106,16 +109,23 @@ pub fn set_dependencies_local_move_stdlib(project_path: &Path) {
...
+    project_folder: &Path,
+    cargo_bin_path: &str,
+) -> anyhow::Result<String> {
     assert!(
         project_folder.exists(),
         "Project folder {:?} does not exist",
         project_folder.display()
     );
-    let output = std::process::Command::new(env!("CARGO_BIN_EXE_dove"))
+    let output = std::process::Command::new(cargo_bin_path)
```
Comment (vitvakatu) [inline dove/src/tests_helper.rs:128]: What's the reason for such change?

Comment (vladimirovmm) [reply dove/src/tests_helper.rs:128]: This variable is only visible in the tests folder

File: dove/src/lib.rs (hunk 1 above)
Comment (vladimirovmm) [reply dove/src/lib.rs:21]: Then the tests will not see this module

File: dove/src/tests_helper.rs [hunk 3]
```diff
@@ -106,16 +109,23 @@ pub fn set_dependencies_local_move_stdlib(project_path: &Path) {
...
 }
 
+/// run dove
 pub fn execute_dove_at(args: &[&str], project_folder: &Path) -> anyhow::Result<()> {
     execute(args, project_folder.to_path_buf())
 }
-pub fn execute_dove_bin_at(args: &[&str], project_folder: &Path) -> anyhow::Result<String> {
+
+/// run bin dove
+pub fn execute_dove_bin_at(
```
Comment (mkurnikov) [inline dove/src/tests_helper.rs:118]: You run `COMMAND_BIN_PATH *ARGS` inside a terminal. So, please, make `bin_path` the first argument.

File: dove/tests/test_cmd_dove_init.rs [hunk 4]
```diff
@@ -161,7 +160,12 @@ fn test_cmd_dove_init_incorrect_repo() {
         "127.0.0.1/api",
         "ftp://demo.ru/api",
     ] {
-        assert!(execute_dove_bin_at(&["dove", "init", "-r", api], &project_folder).is_err());
+        assert!(execute_dove_bin_at(
+            &["dove", "init", "-r", api],
+            &project_folder,
+            env!("CARGO_BIN_EXE_dove")
```
Comment (mkurnikov) [inline dove/tests/test_cmd_dove_init.rs:166]: Could you make this a file constant to reuse with possible future tests?

File: dove/tests/test_cmd_dove_new.rs [hunk 5]
```diff
@@ -162,10 +161,12 @@ fn test_cmd_dove_new_pont_with_incorrect_repo() {
...
         "ftp://demo.ru/api",
     ] {
-        assert!(
-            execute_dove_bin_at(&["dove", "new", &project_name, "-r", api], &base_folder)
-                .is_err()
-        );
+        assert!(execute_dove_bin_at(
+            &["dove", "new", &project_name, "-r", api],
+            &base_folder,
+            env!("CARGO_BIN_EXE_dove")
```
Comment (mkurnikov) [inline dove/tests/test_cmd_dove_new.rs:167]: Same here, file constant.

**Review by mkurnikov — CHANGES_REQUESTED** [review]

File: dove/src/tests_helper.rs (hunk 3 above)
Comment (vladimirovmm) [reply dove/src/tests_helper.rs:118]: > Could you make this a file constant to reuse with possible future tests?

Are you sure? That was your wish

**Review by mkurnikov — APPROVED** [review]

**Pull request merged**